llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -1604,14 +1604,19 @@ bool AIE2InstructionSelector::selectG_AIE_LOAD_UNPACK(
     MachineInstr &UNPACKI, MachineRegisterInfo &MRI) {
   Register LoadResult = (std::next(UNPACKI.uses().begin()))->getReg();
   MachineInstr *LoadOp = getDefIgnoringCopiesAndBitcasts(LoadResult, MRI);
+  bool ShouldAntecipate = false;
 
   assert(LoadOp && "Expected SSA.");
 
   // Do not try to combine if one of the load's defs is used by another
   // instruction between the load and the VUNPACK or if there is a store
   // between the load and the VUNPACK.
-  if (!canDelayMemOp(*LoadOp, UNPACKI, MRI))
-    return false;
+  if (!canDelayMemOp(*LoadOp, UNPACKI, MRI)) {
+    if (canAntecipateOp(*LoadOp, UNPACKI, MRI))
```

Comment:
Nit: ant*i*cipate
I'm wondering if we should name that e.g. `canAdvanceOp` to be more symetric?

Reply:
Sure! Nice suggestion.